libcontainer/nsenter/nsexec.c

```diff
@@ -242,19 +320,75 @@ void nsexec()
 	// We must fork to actually enter the PID namespace, use CLONE_PARENT
 	// so the child can have the right parent, and we don't need to forward
 	// the child's exit code or resend its death signal.
-	child = clone_parent(&env);
+	int child = clone_parent(&env, cloneflags);
 	if (child < 0) {
 		pr_perror("Unable to fork");
 		exit(1);
 	}
+	// if uid_map and gid_map were specified, writes the data to /proc files
+	if (uidmap_start > 0 && uidmap_len > 0) {
+		char buf[PATH_MAX];
+		if (snprintf(buf, sizeof(buf), "/proc/%d/uid_map", child) < 0) {
+			pr_perror("failed to construct uid_map file for %d",
+				  child);
+			exit(1);
+		}
 
-	len = snprintf(buf, sizeof(buf), "{ \"pid\" : %d }\n", child);
+		int fd = open(buf, O_RDWR);
+		writedata(fd, data, uidmap_start, uidmap_len);
+	}
 
-	if (write(pipenum, buf, len) != len) {
+	if (gidmap_start > 0 && gidmap_len > 0) {
+		if (is_setgroup == 1) {
+			char buf[PATH_MAX];
+			if (snprintf
+			    (buf, sizeof(buf), "/proc/%d/setgroups",
+			     child) < 0) {
+				pr_perror
+				    ("failed to construct setgroups file for %d",
+				     child);
```

Comment:
yeah, `indent -linux` makes some weird choices sometimes. I'll refactor the function to cleanup the variable declaration and see how it goes afterwards.